### PR TITLE
WT-6583 Flag history store cursor as WT_CBT_NO_TXN

### PR DESCRIPTION
**Pull request by quchenhao**
In WT-6339, we added the code to flag the history store cursor as WT_CBT_NO_TXN.

```
/*
* Set the flag to stop creating snapshots for history store cursors
*/
F_SET((WT_CURSOR_BTREE *)cursor, WT_CBT_NO_TXN);

```

That part of the code then is removed in WT-6453, causing this issue.

### CODE REVIEW

**Comment by quchenhao** [comment]
@raviprakashgiri29 @agorrod Compared to #5953, this approach is simpler. @raviprakashgiri29 Did you intend to check the isolation level as suggested in the comment in WT-6339?

However, with the change cursor_order test is failing:

```
[2020/08/20 10:38:15.283] ../../../build_posix/gnu-support/test-driver: line 107: 26837 Aborted                 (core dumped) "$@" > $log_file 2>&1
[2020/08/20 10:38:15.312] FAIL: cursor_order
[2020/08/20 10:38:15.312] =========================================================
[2020/08/20 10:38:15.312]    WiredTiger 10.0.0: test/cursor_order/test-suite.log
[2020/08/20 10:38:15.312] =========================================================
[2020/08/20 10:38:15.312] # TOTAL: 1
[2020/08/20 10:38:15.312] # PASS:  0
[2020/08/20 10:38:15.312] # SKIP:  0
[2020/08/20 10:38:15.312] # XFAIL: 0
[2020/08/20 10:38:15.312] # FAIL:  1
[2020/08/20 10:38:15.312] # XPASS: 0
[2020/08/20 10:38:15.312] # ERROR: 0
[2020/08/20 10:38:15.312] .. contents:: :depth: 2
[2020/08/20 10:38:15.312] FAIL: cursor_order
[2020/08/20 10:38:15.312] ==================
[2020/08/20 10:38:15.312] cursor_order: process 26837
[2020/08/20 10:38:15.312]     1: 5 reverse scanners, 1 writers
[2020/08/20 10:38:15.312]  reverse scan thread  0 starting: tid: 26837:0x7f9bdce85700, file: file:cursor_order.000
[2020/08/20 10:38:15.312]  reverse scan thread  1 starting: tid: 26837:0x7f9bd7fff700, file: file:cursor_order.000
[2020/08/20 10:38:15.312]  reverse scan thread  2 starting: tid: 26837:0x7f9bd77fe700, file: file:cursor_order.000
[2020/08/20 10:38:15.312]  reverse scan thread  3 starting: tid: 26837:0x7f9bd6ffd700, file: file:cursor_order.000
[2020/08/20 10:38:15.312]  reverse scan thread  4 starting: tid: 26837:0x7f9bd67fc700, file: file:cursor_order.000
```
I'm yet to understand this.

File: src/history/hs.c [hunk 1]
```diff
@@ -219,6 +219,9 @@ __wt_hs_cursor_open(WT_SESSION_IMPL *session)
     /* History store cursors should always ignore tombstones. */
     F_SET(cursor, WT_CURSTD_IGNORE_TOMBSTONE);
 
+    /* Set the flag to stop creating snapshots for history store cursors. */
+    F_SET((WT_CURSOR_BTREE *)cursor, WT_CBT_NO_TXN);
```
Comment (tetsuo-cpp) [inline src/history/hs.c:223]: Does the cursor opened in `__hs_fixup_out_of_order_from_pos` need this too?

Comment (quchenhao) [reply src/history/hs.c:223]: Yes, that's a good point.

Comment (quchenhao) [reply src/history/hs.c:223]: Also we should flag it as

    /* History store cursors should always ignore tombstones. */
    F_SET(cursor, WT_CURSTD_IGNORE_TOMBSTONE);

Comment (tetsuo-cpp) [reply src/history/hs.c:223]: I don't think that matters. It's just only ever inserting things.

**Comment by quchenhao** [comment]
This is not a correct fix. Close in favour of #5964

**Pull request closed without merging**